kerem-net-nest/src/posts/posts.service.ts

```diff
@@ -0,0 +1,73 @@
+/* eslint-disable @typescript-eslint/no-unsafe-assignment */
+/* eslint-disable @typescript-eslint/no-unnecessary-type-assertion */
+/* eslint-disable prettier/prettier */
+/* eslint-disable @typescript-eslint/no-unsafe-call */
+/* eslint-disable prettier/prettier */
+import { Injectable } from '@nestjs/common';
+import { v4 } from "uuid";
+import { User, UsersService } from '../users/users.service';
+import postsDate from '../Data/posts.json'
+import { CreatePostDto } from './posts.controller';
+import { BadRequestException, NotFoundException } from '@nestjs/common';
+
+
+export interface PostInfo {
+    id: string,
+    userId: string,
+    text: string;
+    comments: {
+        comments:
+        {
+            commentContent: string;
+            userName: string;
+            date: string;
+        }[]
+    };
+    likes: string[];
+    creatorName: string;
+    date: string;
+}
+@Injectable()
+export class PostsService {
+    constructor(private usersService: UsersService) { }
+    private posts: PostInfo[] = postsDate;
+    getAllPosts(): PostInfo[] {
+        console.log(this.posts);
+
+        return this.posts;
+    }
+
+    getPostById(id: string): PostInfo | string {
+        return this.posts.find(post => post.id === id) || 'Post was not found';
+    }
+    createPost(post: CreatePostDto): PostInfo {
+        const newpost: PostInfo = {
+            id: (v4() as string),
+            userId: post.userId,
+            text: post.text,
+            comments: { comments: [] },
+            likes: [],
+            creatorName: post.creatorName,
+            date: new Date().toISOString()
+        };
+        this.posts.push(newpost);
+        return newpost;
+    }
```

Comment:
if you destructure here it would simplify this code
```suggestion
    createPost({userId, text, creatorName}: CreatePostDto): PostInfo {
        const newpost: PostInfo = {
            id: (v4() as string),
            userId,
            text,
            comments: { comments: [] },
            likes: [],
            creatorName
            date: new Date().toISOString()
        };
        this.posts.push(newpost);
        return newpost;
    }
```

Reply:
Absolutely right, it looks much nicer this way